src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs

```diff
@@ -14,13 +14,50 @@ namespace Microsoft.CodeAnalysis
     // all these are just helper methods
     internal partial class Checksum
     {
+        private static readonly Stack<IncrementalHash> s_hashStack = new Stack<IncrementalHash>();
```

Comment:
2 way to leak the incremental hash objects.

1. exception, 2. objectpool dropping the object since pool is full.

for 1, we can add try/finally, for 2, we can change ObjectPool to return bool on Free so that if Free return false, we dispose the object ourselves.